creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1028,15 +1029,59 @@ class SnapbackSM {
       await utils.timeout(SyncMonitoringRetryDelayMs, false)
     }
 
+    const monitoringTimeMs = Date.now() - startTimeMs
+
+    if (finalSecondaryClock > initialSecondaryClock) {
+      secondaryProgressed = true
+    }
+
     /**
      * As Primary for user, record syncRequest outcomes to all secondaries
      * For now, ignore syncRequests where a secondary is behind by more than `MaxExportClockValueRange` as
      *    primary doesn't currently have sufficient tracking
      */
-    if (!additionalSyncRequired && !maxExportRangeExceeded) {
+    let additionalSyncIsRequired
+    if (secondaryCaughtUp) {
+      await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
+      additionalSyncIsRequired = false
+      this.logDebug(`${logMsgString} || Sync completed in ${monitoringTimeMs}ms`)
+
+      // Secondary progressed but has not caught up yet (was behind by more than max export range)
+    } else if (secondaryProgressed) {
       await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
+      additionalSyncIsRequired = true
+      this.log(`${logMsgString} || Secondary progressed from clock ${initialSecondaryClock} to ${finalSecondaryClock} but hasn't caught up. Enqueuing additional syncRequest.`)
+
+      // (1) secondary did not catch up AND (2) secondary did not progress
     } else {
       await SecondarySyncHealthTracker.recordFailure(secondaryUrl, userWallet, syncType)
+
+      // only enqueue an additional sync if we have not already met threshold of daily failures
+      /**
+       * todayCount = secondarysynchealthtracker.getToday's syncFailure count (secondary, wallet, syncType, 'Failure')
+       * if (todayCount > threshold) {
+       *    logMsg secondary met daily failure count threshold
+       * } else {
+       *    additionalSyncIsRequired = true
+       * }
+       */
+      const secondaryUserSyncFailureCountTodayResp = SecondarySyncHealthTracker.getSyncRequestOutcomeMetrics({
+        secondary: secondaryUrl,
+        wallet: userWallet,
+        syncType,
+        outcome: SecondarySyncHealthTracker.Outcomes.FAILURE
+        /* date automatically defaults to today */
+      })
+      // Above resp is a map with a single key-value pair; need to parse out value & handle case where no key-value pair exists
+      // TODO what if key-val pair doesn't exist, or if multiple exist...
+      const secondaryUserSyncFailureCountToday = Object.entries(secondaryUserSyncFailureCountTodayResp)[0][1]
```

Comment:
can we make this a wrapper fn in `SecondarySyncHealthTracker`? that way, the next dev that uses this fn does not need to know how to access the data, but can call a fn in the relevant class

Reply:
yeeup